dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/MetadataInfo.java

```diff
@@ -409,17 +405,7 @@ public boolean hasMethodParameter(String method) {
         }
 
         public String toDescString() {
-            return this.getMatchKey() + getMethodSignaturesString() + new TreeMap<>(getParams());
-        }
-
-        private String getMethodSignaturesString() {
-            SortedSet<String> methodStrings = new TreeSet();
-
-            Method[] methods = ClassUtils.forName(name).getMethods();
-            for (Method method : methods) {
-                methodStrings.add(method.toString());
-            }
-            return methodStrings.toString();
+            return this.getMatchKey() + name + group + version + path + protocol + new TreeMap<>(getParams());
```

Comment:
The following two parts have the same meaning：

`name + group + version + path + protocol`

`getServiceKey()`

Reply:
done, `getMatchKey()` contains `name + group + version + protocol`